### PR TITLE
Search form

### PR DESCRIPTION
**Pull request by dsbegnoche**
Adding forms search parameter to filter forms by environment id, as discussed in https://github.com/oneblink/sdk-dotnet/issues/94

### CODE REVIEW

File: docs/forms-client.md [hunk 1]
```diff
@@ -132,6 +134,7 @@ OneBlink.SDK.Model.FormsSearchResult response = await formsClient.Search(isAuthe
 | `isAuthenticated` | Yes      | `bool?`  | Return authenticated forms or unauthenticated forms. If null provided, all forms will be returned.                                           |
 | `isPublished`     | Yes      | `bool?`  | Return published forms or unpublished forms. If null provided, all forms will be returned.                                                   |
 | `name`            | Yes      | `string` | Search on the name property of a form. Can be a prefix, suffix or partial match. If null or whitespace provided, all forms will be returned. |
+| `formsAppEnvironment`            | No      | `string` | Return only forms for a specific app environment. |
```
Comment (mymattcarroll) [inline docs/forms-client.md:137]:
```suggestion
| `formsAppEnvironment`            | No      | `long` | Return only forms for a specific app environment. |
```

File: docs/forms-client.md [hunk 2]
```diff
@@ -122,7 +122,9 @@ A `FormSubmission<T>` class
 bool? isAuthenticated = null;
 bool? isPublished = null;
 string name = null;
-OneBlink.SDK.Model.FormsSearchResult response = await formsClient.Search(isAuthenticated, isPublished, name);
+int? formsAppEnvironment = null;
```
Comment (mymattcarroll) [inline docs/forms-client.md:125]:
```suggestion
long? formsAppEnvironmentId = null;
```

File: docs/forms-client.md [hunk 3]
```diff
@@ -122,7 +122,9 @@ A `FormSubmission<T>` class
...
+
+OneBlink.SDK.Model.FormsSearchResult response = await formsClient.Search(isAuthenticated, isPublished, name, formsAppEnvironment);
```
Comment (mymattcarroll) [inline docs/forms-client.md:127]:
```suggestion
OneBlink.SDK.Model.FormsSearchResult response = await formsClient.Search(isAuthenticated, isPublished, name, formsAppEnvironmentId);
```

File: OneBlink.SDK/FormsClient.cs [hunk 4]
```diff
@@ -307,4 +317,9 @@ public async Task<SubmissionDataUrl> GenerateSubmissionDataUrl(long formId, stri
             return url;
         }
     }
+
+    public class WebhookRequest
+    {
+        public string callbackUrl { get; set; }
+    }
```
Comment (mymattcarroll) [inline OneBlink.SDK/FormsClient.cs:324]: While we understand why you have added this class to use as a POST request payload, it does not belong in this file. If you would like it in this code base, move it to its own file under the `models` directory.

File: OneBlink.SDK/FormsClient.cs [hunk 5]
```diff
@@ -47,7 +47,7 @@ public async Task<FormSubmission<T>> GetFormSubmission<T>(long formId, string su
             return await GetFormSubmission<T>(formRetrievalData);
         }
 
-        public async Task<FormsSearchResult> Search(bool? isAuthenticated, bool? isPublished, string name)
+        public async Task<FormsSearchResult> Search(bool? isAuthenticated, bool? isPublished, string name, int? formsAppEnvironment = null)
```
Comment (mymattcarroll) [inline OneBlink.SDK/FormsClient.cs:50]:
```suggestion
        public async Task<FormsSearchResult> Search(bool? isAuthenticated, bool? isPublished, string name, long? formsAppEnvironmentId = null)

```

File: OneBlink.SDK/FormsClient.cs [hunk 6]
```diff
@@ -70,6 +70,15 @@ public async Task<FormsSearchResult> Search(bool? isAuthenticated, bool? isPubli
                 }
                 queryString += "name=" + name;
             }
+            if (formsAppEnvironment.HasValue)
```
Comment (mymattcarroll) [inline OneBlink.SDK/FormsClient.cs:73]:
```suggestion
            if (formsAppEnvironmentId.HasValue)

```

File: OneBlink.SDK/FormsClient.cs [hunk 7]
```diff
@@ -70,6 +70,15 @@ public async Task<FormsSearchResult> Search(bool? isAuthenticated, bool? isPubli
                 }
                 queryString += "name=" + name;
             }
+            if (formsAppEnvironment.HasValue)
+            {
+                if (queryString != string.Empty)
+                {
+                    queryString += "&";
+                }
+                queryString += "formsAppEnvironmentId=" + formsAppEnvironment.Value;
```
Comment (mymattcarroll) [inline OneBlink.SDK/FormsClient.cs:79]:
```suggestion
                queryString += "formsAppEnvironmentId=" + formsAppEnvironmentId.Value;

```

**Review by mymattcarroll — CHANGES_REQUESTED** [review]
Thank you for your contribution.

There are just a few things that will need to be addressed before we can approve and merge this Pull Request.

File: OneBlink.SDK/FormsClient.cs (hunk 4 above)
Comment (dsbegnoche) [reply OneBlink.SDK/FormsClient.cs:324]: This was accidental, We also need to make this request, but I believe we will be doing this outside of the SDK. Reverting.

File: docs/forms-client.md (hunk 1 above)
Comment (dsbegnoche) [reply docs/forms-client.md]: Updated to have Id suffix and type of `long?`

**Comment by dsbegnoche** [comment]
Thank you for the quick review. Updated PR with requested changes.

**Review by mymattcarroll — APPROVED** [review]

**Comment by mymattcarroll** [comment]
@dsbegnoche Thank you for your contribution. We currently have unreleased features in `master` and cannot release them until the associated server work is done in the OneBlink product. We will publish a beta release soon with your changes, however, please be aware that a beta release cannot be considered stable and may not function exactly as intended.

**Pull request merged**